pkg/apis/kops/validation/legacy.go

```diff
@@ -291,6 +291,36 @@ func ValidateCluster(c *kops.Cluster, strict bool) field.ErrorList {
 					}
 				}
 			}
+
+			// @ check that NodeLocalDNS addon is configured correctly
+			if c.Spec.KubeDNS.NodeLocalDNS != nil && c.Spec.KubeDNS.NodeLocalDNS.Enabled {
+				if c.Spec.KubeDNS.Provider != "CoreDNS" {
+					allErrs = append(allErrs, field.Forbidden(fieldSpec.Child("kubeDNS", "provider"), "KubeDNS provider must be set to CoreDNS if NodeLocalDNS addon is enabled"))
+				}
+
+				if c.Spec.KubeDNS.NodeLocalDNS.LocalIP == "" {
+					if c.Spec.Kubelet != nil && c.Spec.Kubelet.ClusterDNS != c.Spec.KubeDNS.NodeLocalDNS.LocalIP {
```

Comment:
@olemarkus need to change this. Just realized that this does not make the sense and breaks also the case for `iptables`.

Reply:
Improved entire validation logic.

Reply:
Writing test for the `validateNodeLocalDNS` func